src/components/remote_control/test/commands/button_press_request_test.cc

```diff
@@ -0,0 +1,270 @@
+#include "remote_control/message_helper.h"
+#include "remote_control/rc_command_factory.h"
+#include "remote_control/event_engine/event_dispatcher.h"
+#include "functional_module/function_ids.h"
+#include "include/mock_service.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+
+using functional_modules::MobileFunctionID;
+using application_manager::ServicePtr;
+
+using application_manager::MockService;
+using test::components::remote_control_test::MockApplication;
+
+using ::testing::_;
+using ::testing::Mock;
+using ::testing::NiceMock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::SaveArg;
+using ::application_manager::Message;
+using ::application_manager::MessageType;
+using ::protocol_handler::MessagePriority;
+using remote_control::RemotePluginInterface;
+using remote_control::MessageHelper;
+using namespace remote_control;
+
+namespace {
+const int kModuleId = 153;
+const std::string kCorrectMobileRequest =
+    "{\"moduleType\":\"CLIMATE\",\"buttonName\":\"AC\",\"buttonPressMode\":"
+    "\"SHORT\"}";
+const std::string kWrongMobileRequest =
+    "{\"moduleType\":\"RADIO\",\"buttonName\":\"AC\",\"buttonPressMode\":"
+    "\"SHORT\"}";
+const std::string kValidHmiResponse =
+    "{\"result\":{\"code\":0,\"method\":\"Buttons.ButtonPress\"},\"id\":31,"
+    "\"jsonrpc\":\"2.0\"} ";
+const std::string KReadOnlyHmiResponse =
+    "{\"jsonrpc\":\"2.0\",\"error\":{\"code\":26,\"data\":{\"method\":"
+    "\"Buttons.ButtonPress\"},\"message\":\"Read only parameters "
+    "received\"},\"id\":31}";
+const uint32_t kAppId = 11u;
+}
+
+namespace test {
+namespace components {
+namespace remote_control_test {
+namespace button_press_request_test {
+
+class ButtonPressRequestTest : public ::testing::Test {
+ public:
+  ButtonPressRequestTest()
+      : mock_service_(utils::MakeShared<NiceMock<MockService> >())
+      , mock_app_(utils::MakeShared<NiceMock<MockApplication> >())
+      , rc_app_extention_(
+            utils::MakeShared<remote_control::RCAppExtension>(kModuleId)) {
+    ON_CALL(mock_module_, service()).WillByDefault(Return(mock_service_));
+    ON_CALL(*mock_service_, GetApplication(kAppId))
+        .WillByDefault(Return(mock_app_));
+    EXPECT_CALL(mock_module_, event_dispatcher())
+        .WillRepeatedly(ReturnRef(event_dispatcher_));
+    ServicePtr exp_service(mock_service_);
+    mock_module_.set_service(exp_service);
+  }
+
+  remote_control::request_controller::MobileRequestPtr CreateCommand(
+      application_manager::MessagePtr msg) {
+    return remote_control::RCCommandFactory::CreateCommand(msg, mock_module_);
+  }
+
+  application_manager::MessagePtr CreateBasicMessage() {
+    application_manager::MessagePtr message = utils::MakeShared<Message>(
+        MessagePriority::FromServiceType(protocol_handler::ServiceType::kRpc));
+    message->set_function_id(MobileFunctionID::BUTTON_PRESS);
+    message->set_function_name("ButtonPress");
```

Comment:
@LitvinenkoIra Please use `GetMobileAPIName(MobileFunctionID func_id)` instead of hardcoding